src/Features/VisualBasic/Portable/CodeFixes/MoveToTopOfFile/MoveToTopOfFileCodeFixProvider.vb

```diff
@@ -144,9 +144,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeFixes.MoveToTopOfFile
             }
         End Function
 
-        Private Function FindLastContiguousStatement(nodes As IEnumerable(Of SyntaxNode)) As Integer
+        Private Function FindLastContiguousStatement(nodes As IEnumerable(Of SyntaxNode), trivia As IEnumerable(Of SyntaxTrivia)) As Integer
             If Not nodes.Any() Then
-                Return 0
+                If Not trivia.Any(Function(t) t.IsKind(SyntaxKind.CommentTrivia)) Then
```

Comment:
To avoid enumerating the sequence twice (once for `Any()` and once for `Last()`) what about something like:

``` VB
Dim lastTrivia = trivia.LastOrDefault(Function(t) ...)
If lastTrivia Is Nothing Then
    Return 0
Else
    Return lastTrivia.GetLocation()...
End If
```